### PR TITLE
Haogang's assignment 7!

### PR DESCRIPTION
**Pull request by rhg950510**
I'm curious about the perfect ROC :)

### CODE REVIEW

**Comment by rhg950510** [comment]
Please the Update version. I did not save the Rmd file on the original one.

**Comment by rhg950510** [comment]
Please see the update version. I did not save to the original Rmd file.

File: Assignment7.Rmd [hunk 1]
```diff
@@ -11,62 +11,88 @@ In the following assignment you will be looking at data from an one level of an
...
+df %>%
+  keep(is.numeric) %>% 
+  gather() %>% 
+  ggplot(aes(value)) +
+    facet_wrap(~ key, scales = "free") +
+    geom_histogram()
 #Then visualize the relationships between variables
-
+pairs(df)
+cor(df %>%keep(is.numeric))
```
Comment (charles-lang) [inline Assignment7.Rmd]: Good work 👍 Excellent visualizations!

File: Assignment7.Rmd [hunk 2]
```diff
@@ -11,62 +11,88 @@ In the following assignment you will be looking at data from an one level of an
...
+c.tree <- rpart(level.up~post.test.score+messages+av.assignment.score,data=df,method = "class")
 #Plot and generate a CP table for your tree 
-
+printcp(c.tree)
+rpart.plot(c.tree)
 #Generate a probability value that represents the probability that a student levels up based your classification tree 
-
-D1$pred <- predict(rp, type = "prob")[,2]#Last class we used type = "class" which predicted the classification for us, this time we are using type = "prob" to see the probability that our classififcation is based on.
+df$pred <- predict(c.tree, type = "prob")[,2]#Last class we used type = "class" which predicted the classification for us, this time we are using type = "prob" to see the probability that our classififcation is based on.
 ```
```
Comment (charles-lang) [inline Assignment7.Rmd]: Good work 👍

File: Assignment7.Rmd [hunk 3]
```diff
@@ -11,62 +11,88 @@ In the following assignment you will be looking at data from an one level of an
...
 #Now repeat this process, but using the variables you did not use for the previous model and compare the plots & results of your two models. Which one do you think was the better model? Why?
+c.tree2 <- rpart(level.up ~ pre.test.score + forum.posts + messages,method="class", data = df)
+printcp(c.tree2)
+rpart.plot(c.tree2)
+df$pred2 <- predict(c.tree2, type = "prob")[,2]
+pred.detail2 <- prediction(df$pred2, df$level.up) 
+plot(performance(pred.detail2, "tpr", "fpr"))
+abline(0, 1, lty = 2)
+unlist(slot(performance(pred.detail2,"auc"), "y.values"))
+#Therefore, The first tree is the better model with greater AUC.
```
Comment (charles-lang) [inline Assignment7.Rmd]: It is always a good idea to explain what you mean by better.

File: Assignment7.Rmd [hunk 4]
```diff
@@ -11,62 +11,88 @@ In the following assignment you will be looking at data from an one level of an
...
 
-D1$recall.model1 <- 
+#Now generate three diagnostics:
+accuracy.model1 <- mean(ifelse(df$level.up == df$threshold.pred1, 1, 0))
+df$truepos.model1 <- ifelse(df$level.up == "yes" & df$threshold.pred1 == "yes", 1, 0)
+df$falsepos.model1 <- ifelse(df$level.up == "no" & df$threshold.pred1 == "yes", 1,0)
+df$falseneg.model1 <- ifelse(df$level.up == "yes" & df$threshold.pred1 == "no", 1,0)
+precision.model1 <- sum(df$truepos.model1)/(sum(df$truepos.model1) + sum(df$falsepos.model1))
+precision.model1
+recall.model1 <- sum(df$truepos.model1)/(sum(df$truepos.model1) + sum(df$falseneg.model1))
```
Comment (charles-lang) [inline Assignment7.Rmd]: Great!

**Pull request closed without merging**